src/main/java/domain/lotto/service/LottoService.java

```diff
@@ -0,0 +1,67 @@
+package domain.lotto.service;
+
+import domain.common.Money;
+import domain.lotto.LottoResult;
+import domain.lotto.LottoTicket;
+import domain.lotto.Prize;
+import domain.lotto.dto.MatchResult;
+import domain.lotto.dto.StatistsDto;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LottoService {
+
+    public StatistsDto makeStatistics(Money myMoney, LottoResult lottoResult, List<LottoTicket> lottoTickets) {
+        Map<Prize, Integer> prizes = getPrizes();
+        
+        lottoTickets.forEach(lottoTicket -> {
+            Prize prize = lottoResult.findPrize(lottoTicket);
+            prizes.put(prize, prizes.getOrDefault(prize, 0) + 1);
+        });
+        
+        List<MatchResult> matchResults = getMathResult(prizes);
+        Money total = calculateTotalMoney(prizes);
+        double profitRate = (double) total.getAmount() / myMoney.getAmount();
+        profitRate = Math.floor(profitRate * 100) / 100;
+        return new StatistsDto(matchResults, profitRate);
+    }
+
+    private Map<Prize, Integer> getPrizes() {
+        Map<Prize, Integer> prizes = new LinkedHashMap<>();
+        prizes.put(Prize.FOURTH, 0);
+        prizes.put(Prize.THIRD, 0);
+        prizes.put(Prize.SECOND, 0);
+        prizes.put(Prize.FIRST, 0);
+        return prizes;
+    }
+
+    private List<MatchResult> getMathResult(Map<Prize, Integer> prizes) {
+        List<MatchResult> matchResults = new ArrayList<>();
+        for (Map.Entry<Prize, Integer> prizeIntegerEntry : prizes.entrySet()) {
+            Prize prize = prizeIntegerEntry.getKey();
+            Integer match = prize.getMatch();
+            Money money = prize.getMoney();
+            if (prize == Prize.LOSING_TICKET) continue;
+
+            Integer count = prizeIntegerEntry.getValue();
+            matchResults.add(new MatchResult(match, money.getAmount(), count));
+        }
+        return matchResults;
+    }
```

Comment:
```suggestion
    private List<MatchResult> getMathResult(Map<Prize, Integer> prizes) {
        return prizes.keySet().stream()
                .filter(k -> k != Prize.LOSING_TICKET)
                .map(k -> new MatchResult(k.getMatch(), k.getMoney().getAmount(), prizes.get(k)))
                .toList();
    }
```
이 부분은 아래와 같이 수정하는 것이 더 가독성이 좋을 것 같습니다!
더 가독성을 좋게 하려면 MatchResult 내에 Prize와 Map<Prize, Integer>를 파라미터로 받는 생성자나 정적팩토리메소드가 있다면 더 간결하게 수정할 수 있을 것 같습니다

Reply:
크~ 저도 stream 쓰려고 노력은 하는데, map에 대해서도 좋은 스트림 예제를 보여주시는군요 감사합니다!